runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/PTransformTranslation.java

```diff
@@ -124,6 +124,15 @@
   public static final String SPLITTABLE_PROCESS_SIZED_ELEMENTS_AND_RESTRICTIONS_URN =
       "beam:transform:sdf_process_sized_element_and_restrictions:v1";
 
+  public static final String REQUIRES_STATEFUL_PROCESSING_URN =
+      getUrn(RunnerApi.StandardRequirements.Enum.REQUIRES_STATEFUL_PROCESSING);
```

Comment:
Please define string constants and then validate that they match in the static initializer. This allows for the string constants to be used in switch/case statements since they are compile time constants.

I have pr/11162 which duplicates this effort, which ever goes in second needs to get updated.

Reply:
That's unfortunate, but I see the pattern. Have to be vigilant to prevent bugs. (Unlikely that these'll be used in switch statements, but consistency is good.)